src/lib/dom-api-mutation.html

```diff
@@ -0,0 +1,219 @@
+            h = n.__observeNodesMap.get(this);
+            if (h) {
+              Polymer.dom(n).unobserveNodes(h);
+              n.__observeNodesMap.delete(this);
+            }
+          }
+        }
+      },
+
+      _isContent: function(node) {
+        return (node.localName === 'content');
+      },
+
+      _callListeners: function() {
+        var o$ = this._listeners;
+        var nodes = this.domApi.getEffectiveChildNodes();
+        for (var i=0, o; (i < o$.length) && (o=o$[i]); i++) {
+          var info = this._generateListenerInfo(o, nodes);
+          if (info || o._alwaysCallListener) {
+            this._callListener(o, info);
+          }
+        }
+      },
+
+      _generateListenerInfo: function(listener, newNodes) {
+        var oldNodes = listener._nodes;
+        var info = {
+          target: this.node,
+          addedNodes: [],
+          removedNodes: []
+        };
+        var splices = Polymer.ArraySplice.calculateSplices(newNodes, oldNodes);
+        // process removals
+        for (var i=0, s; (i<splices.length) && (s=splices[i]); i++) {
+          for (var j=0, n; (j < s.removed.length) && (n=s.removed[j]); j++) {
+            info.removedNodes.push(n);
+          }
+        }
+        // process adds
+        for (var i=0, s; (i<splices.length) && (s=splices[i]); i++) {
+          for (var j=s.index; j < s.index + s.addedCount; j++) {
+            info.addedNodes.push(newNodes[j]);
+          }
+        }
+        // update cache
+        listener._nodes = newNodes;
+        if (info.addedNodes.length || info.removedNodes.length) {
+          return info;
+        }
+      },
+
+      _callListener: function(listener, info) {
+        return listener.fn.call(this.node, info);
+      }
+
+    };
+
+    if (Settings.useShadow) {
+
+      var baseSetup = DomApi.Mutation.prototype._setup;
+      var baseCleanup = DomApi.Mutation.prototype._cleanup;
+
+      var beforeCallListeners = DomApi.Mutation.prototype._beforeCallListeners;
+
+      Polymer.Base.extend(DomApi.Mutation.prototype, {
+        
+        _setup: function() {
+          if (!this._observer) {
+            var self = this;
+            this._mutationHandler = function(mxns) {
+              self._scheduleNotify();
+            };
+            this._observer = new MutationObserver(this._mutationHandler);
+            this._preflush = this._flush.bind(this);
+            Polymer.dom.addPreflush(this._preflush);
+            // NOTE: subtree true is way too aggressive, but it easily catches
+            // attribute changes on children. These changes otherwise require
+            // attribute observers on every child. Testing has shown this
+            // approach to be more efficient. 
+            // TODO(sorvell): do we still need to include an option to defeat
+            // attribute tracking?
+            this._observer.observe(this.node, {
+              childList: true,
+              attributes: true,

```

Comment:
Sucks that there's no `childList` + `attributes` combo that actually means "any attribute changes on my childList".

So any change (add, remove, attribute change) to any of its light children descendants will incur the cost of an `_unobserveContentElements` iteration (once) and a levenshtein (per listener).

Seems potentially expensive enough to keep the attributes option, since it allows you to remove _both_ `attributes` + `subtree`.

Is there an option to auto-enable attributes if you encounter a `select` on a `<content>` while observing?  Or is that too complicated?